### ISSUE TITLE
API response for new MARKET order is sending wrong execution price

### ISSUE THREAD

**Issue by vzubach** [mxcdevelop/mexc-api-sdk#77]
I send a request to open a new order on the market with quoteOrderQty = 6 USDT 

```
tstmp=$(date +%s%N | cut -b1-13)
querystr="symbol=KASUSDT&side=BUY&type=MARKET&quoteOrderQty=6&&timestamp=$tstmp"
sign=$(echo -n $querystr | openssl dgst -sha256 -hmac "$SKey" | awk '{print $2}')
NewOrder=$(curl -sLH "x-mexc-apikey: $AKey" -H "Content-Type: application/json" -X POST "$url/api/v3/order?$querystr&signature=$sign")
echo "$NewOrder" | jq -r .
{
  "symbol": "KASUSDT",
  "orderId": "C02__427495284067573760015",
  "orderListId": -1,
  "price": "0.183503",   <----- WRONG PRICE! The actual strike price was 0.175137
  "origQty": "32.69",
  "type": "MARKET",
  "side": "BUY",
  "transactTime": 1717705054066
}
```
Query Order with  "orderId": "C02__427495284067573760015" response:
```
{
  "symbol": "KASUSDT",
  "orderId": "C02__427495284067573760015",
  "orderListId": -1,
  "clientOrderId": null,
  "price": "0.183503",   <----- Wrong executed price. The actual strike price was 0.175137 = (5.72522853 / 32.69)
  "origQty": "32.69",
  "executedQty": "32.69",
  "cummulativeQuoteQty": "5.72522853",    <----- Real executed quantity
  "status": "FILLED",
  "timeInForce": null,
  "type": "MARKET",
  "side": "BUY",
  "stopPrice": null,
  "icebergQty": null,
  "time": 1717705054000,
  "updateTime": 1717705054000,
  "isWorking": true,
  "origQuoteOrderQty": "5.99871307"   <----- Actual purchase quantity requested
}
```

**Comment by khoangkhac118** [comment]
> I send a request to open a new order on the market with quoteOrderQty = 6 USDT 
> 
> ```
> tstmp=$(date +%s%N | cut -b1-13)
> querystr="symbol=KASUSDT&side=BUY&type=MARKET&quoteOrderQty=6&&timestamp=$tstmp"
> sign=$(echo -n $querystr | openssl dgst -sha256 -hmac "$SKey" | awk '{print $2}')
> NewOrder=$(curl -sLH "x-mexc-apikey: $AKey" -H "Content-Type: application/json" -X POST "$url/api/v3/order?$querystr&signature=$sign")
> echo "$NewOrder" | jq -r .
> {
>   "symbol": "KASUSDT",
>   "orderId": "C02__427495284067573760015",
>   "orderListId": -1,
>   "price": "0.183503",   <----- WRONG PRICE! The actual strike price was 0.175137
>   "origQty": "32.69",
>   "type": "MARKET",
>   "side": "BUY",
>   "transactTime": 1717705054066
> }
> ```
> Query Order with  "orderId": "C02__427495284067573760015" response:
> ```
> {
>   "symbol": "KASUSDT",
>   "orderId": "C02__427495284067573760015",
>   "orderListId": -1,
>   "clientOrderId": null,
>   "price": "0.175137",   
>   "origQty": "32.69",
>   "executedQty": "32.69",
>   "cummulativeQuoteQty": "5.72522853",    <----- Real executed quantity
>   "status": "FILLED",
>   "timeInForce": null,
>   "type": "MARKET",
>   "side": "BUY",
>   "stopPrice": null,
>   "icebergQty": null,
>   "time": 1717705054000,
>   "updateTime": 1717705054000,
>   "isWorking": true,
>   "origQuoteOrderQty": "5.99871307"   <----- Actual purchase quantity requested
> }
> ```
>

**Comment by khoangkhac118** [comment]
> I send a request to open a new order on the market with quoteOrderQty = 6 USDT 
> 
> ```
> tstmp=$(date +%s%N | cut -b1-13)
> querystr="symbol=KASUSDT&side=BUY&type=MARKET&quoteOrderQty=6&&timestamp=$tstmp"
> sign=$(echo -n $querystr | openssl dgst -sha256 -hmac "$SKey" | awk '{print $2}')
> NewOrder=$(curl -sLH "x-mexc-apikey: $AKey" -H "Content-Type: application/json" -X POST "$url/api/v3/order?$querystr&signature=$sign")
> echo "$NewOrder" | jq -r .
> {
>   "symbol": "KASUSDT",
>   "orderId": "C02__427495284067573760015",
>   "orderListId": -1,
>   "price": "0.183503",   <----- WRONG PRICE! The actual strike price was 0.175137
>   "origQty": "32.69",
>   "type": "MARKET",
>   "side": "BUY",
>   "transactTime": 1717705054066
> }
> ```
> Query Order with  "orderId": "C02__427495284067573760015" response:
> ```
> {
>   "symbol": "KASUSDT",
>   "orderId": "C02__427495284067573760015",
>   "orderListId": -1,
>   "clientOrderId": null,
>   "price": "0.183503",   <----- Wrong executed price. The actual strike price was 0.175137 = (5.72522853 / 32.69)
>   "origQty": "32.69",
>   "executedQty": "32.69",
>   "cummulativeQuoteQty": "5.72522853",    <----- Real executed quantity
>   "status": "FILLED",
>   "timeInForce": null,
>   "type": "MARKET",
>   "side": "BUY",
>   "stopPrice": null,
>   "icebergQty": null,
>   "time": 1717705054000,
>   "updateTime": 1717705054000,
>   "isWorking": true,
>   "origQuoteOrderQty": "5.99871307"   <----- Actual purchase quantity requested
> }
> ```
>

**Comment by sergey-crusher** [comment]
> > I send a request to open a new order on the market with quoteOrderQty = 6 USDT
> > ```
> > tstmp=$(date +%s%N | cut -b1-13)
> > querystr="symbol=KASUSDT&side=BUY&type=MARKET&quoteOrderQty=6&&timestamp=$tstmp"
> > sign=$(echo -n $querystr | openssl dgst -sha256 -hmac "$SKey" | awk '{print $2}')
> > NewOrder=$(curl -sLH "x-mexc-apikey: $AKey" -H "Content-Type: application/json" -X POST "$url/api/v3/order?$querystr&signature=$sign")
> > echo "$NewOrder" | jq -r .
> > {
> >   "symbol": "KASUSDT",
> >   "orderId": "C02__427495284067573760015",
> >   "orderListId": -1,
> >   "price": "0.183503",   <----- WRONG PRICE! The actual strike price was 0.175137
> >   "origQty": "32.69",
> >   "type": "MARKET",
> >   "side": "BUY",
> >   "transactTime": 1717705054066
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Query Order with  "orderId": "C02__427495284067573760015" response:
> > ```
> > {
> >   "symbol": "KASUSDT",
> >   "orderId": "C02__427495284067573760015",
> >   "orderListId": -1,
> >   "clientOrderId": null,
> >   "price": "0.175137",   
> >   "origQty": "32.69",
> >   "executedQty": "32.69",
> >   "cummulativeQuoteQty": "5.72522853",    <----- Real executed quantity
> >   "status": "FILLED",
> >   "timeInForce": null,
> >   "type": "MARKET",
> >   "side": "BUY",
> >   "stopPrice": null,
> >   "icebergQty": null,
> >   "time": 1717705054000,
> >   "updateTime": 1717705054000,
> >   "isWorking": true,
> >   "origQuoteOrderQty": "5.99871307"   <----- Actual purchase quantity requested
> > }
> > ```

This error remains
New appeared on 07/20/2024
"origQty" began to always return 0, although in fact everything turned out fine

{symbol=BTCUSDC, orderId=C02__443195790819696641107, orderListId=-1, price=69877.5, **origQty=0**, type=MARKET, side=BUY, transactionTime=1721448346577}

**Comment by thxsh** [comment]
same issue here, seems its picking price on the edge of order book, so highest for buy and lowest for sell. 
will this be fixed @MXCDev ?

**Comment by 0xAtasoy** [comment]
the problem is still not solved. it returns "origQty" correctly but "price" is incorrect. as @thxsh said , its picking price on the edge of order book.

**Comment by amjad12bar** [comment]
is problem still not solved ?

**Comment by amjad12bar** [comment]
I have now in 2025 the same problem

**Comment by soutruoshim** [comment]
yes i got this issue in 25/05/2025 too, how to resovle this problem.

**Comment by sergey-crusher** [comment]
I save the previous amount of all coins and compare it with how much is left after the transaction.
